### PR TITLE
Add support for modified huber loss in SGD

### LINKED ISSUE #413: PyTorch `predict_proba` does not match original SGDClassifier model
I have been trialing using hummingbird to convert a [SGDClassifier ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) with  modified_huber loss to a PyTorch model. 

The predictions of the converted model match the original (and it was amazingly easy to convert, really awesome). 

However, calling `predict_proba` the output does not match. Its a factor of 10 smaller and loosely correlated with the originals model confidence. 
plot

**Is it possible to replicate both the classification as well as the confidence in the PyTorch model?**

As a minimal example, take this code:
```
import numpy as np
from sklearn.linear_model import SGDClassifier
from hummingbird.ml import convert

X = np.array([[-0.5, -1], [-1, -1], [-0.1, -0.1], [0.1, -0.2],
              [0.5, 1], [1, 1], [0.1, 0.1], [-0.1, 0.2]])
Y = np.array([1, 1, 1, 1, 2, 2, 2, 2])

clf = SGDClassifier(loss='modified_huber', max_iter=1000, tol=1e-3)
clf.fit(X, Y)

# Use Hummingbird to convert the model to PyTorch
model = convert(clf, 'pytorch')

inputs = [[-1, -1], [1, 1], [-0.2, 0.1], [0.2, -0.1]]
print('Original model')
print("clf.predict(inputs) = {}".format(clf.predict(inputs)))
print("clf.predict_proba(inputs) = \n{}".format(clf.predict_proba(inputs)))

print('Converted pytorch model')
print("model.predict(inputs) = {}".format(model.predict(inputs)))
print("model.predict_proba(inputs) = \n{}".format(model.predict_proba(inputs)))

```

The confidence outputed for the converted model is different every time but can be:
```
Original model
clf.predict(inputs) = [1 2 2 1]
clf.predict_proba(inputs) = 
[[1. 0.]
 [0. 1.]
 [0. 1.]
 [1. 0.]]
Converted pytorch model
model.predict(inputs) = [1 2 2 1]
model.predict_proba(inputs) = 
[[1.         0.        ]
 [0.         1.        ]
 [0.         1.        ]
 [0.7743539  0.22564608]]
```

### PR DESCRIPTION
**Pull request by interesaaat**
Closed #413.

### CODE REVIEW

File: hummingbird/ml/operator_converters/sklearn/linear.py [hunk 1]
```diff
@@ -39,15 +39,23 @@ def convert_sklearn_linear_model(operator, device, extra_config):
...
             multi_class = "ovr"
         else:
             multi_class = "multinomial"
+    if hasattr(operator.raw_operator, "loss"):
+        loss = operator.raw_operator.loss
+        assert (
+            loss == "log" or loss == "modified_huber" or loss == "squared_hinge"
+        ), "predict_proba for linear model currently does not support {}. Please fill an issue at https://github.com/microsoft/hummingbird".format(
+            loss
+        )
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/sklearn/linear.py:58]: Should we add a check here for other such settings that we don't support?

ex : "[The possible options ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html)are ‘hinge’, ‘log’, ‘modified_huber’, ‘squared_hinge’, ‘perceptron’, or a regression loss: ‘squared_loss’, ‘huber’, ‘epsilon_insensitive’, or ‘squared_epsilon_insensitive’."

File: hummingbird/ml/operator_converters/_linear_implementations.py [hunk 2]
```diff
@@ -32,15 +33,31 @@ def __init__(self, coefficients, intercepts, device, classes=[0], multi_class=No
...
             if not self.binary_classification:
-                output /= torch.sum(output, dim=1, keepdim=True)
+                if self.loss == "modified_huber":
+                    # This loss might assign zero to all classes, which doesn't
+                    # normalize neatly; work around this to produce uniform
+                    # probabilities.
+                    prob_sum = torch.sum(output, dim=1, keepdim=False)
+                    all_zero = prob_sum == 0
+                    if torch.any(all_zero):
+                        output[all_zero, :] = 1
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/_linear_implementations.py:56]: Can we add a test for lines 56-57?

File: tests/test_sparkml_pipeline.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
 class TestSparkMLPipeline(unittest.TestCase):
     @unittest.skipIf(not sparkml_installed(), reason="Spark-ML test requires pyspark")
-    @unittest.skipIf(LooseVersion(torch.__version__) < LooseVersion("1.6.0"), reason="Spark-ML test requires torch >= 1.16.0")
```
Comment (ksaur) [inline tests/test_sparkml_pipeline.py]: Is this a copy-paste err?

File: hummingbird/ml/operator_converters/sklearn/linear.py (hunk 1 above)
Comment (interesaaat) [reply hummingbird/ml/operator_converters/sklearn/linear.py:58]: Workin on this!

File: hummingbird/ml/operator_converters/_linear_implementations.py (hunk 2 above)
Comment (interesaaat) [reply hummingbird/ml/operator_converters/_linear_implementations.py:56]: Done!

File: tests/test_sparkml_pipeline.py (hunk 3 above)
Comment (interesaaat) [reply tests/test_sparkml_pipeline.py]: No no I fixed a typo here.

Comment (ksaur) [reply tests/test_sparkml_pipeline.py]: Ok but it is confusing that you are fixing a SparkML comment in a PR related to SGD. Can you please put that in a different PR?
